.github/workflows/wheel_macos_x86_64.yml

```diff
@@ -82,7 +83,7 @@ jobs:
                           -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
                           -DCMAKE_CXX_STANDARD=20 \
                           -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
-                          -DCMAKE_CXX_COMPILER=$(brew --prefix llvm)/bin/clang++ \
+                          -DCMAKE_CXX_COMPILER=g++ \
```

Comment:
So we build with `g++` now?

Reply:
`g++` on MacOS maps to the internal clang compiler. I set it here to avoid `clang` being captured by the on-path brew installed version.